node-installer/integration-test-kind.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+set -euo pipefail
+
+: ${IMAGE_NAME:=ghcr.io/spinkube/containerd-shim-spin/node-installer:dev}
+
+echo "=== Step 1: Create a kind cluster ==="
+if kind get clusters | grep -q "spin-test"; then
+  echo "Deleting existing cluster..."
+  kind delete cluster --name spin-test
+fi
+
+echo "Creating kind cluster..."
+kind create cluster --config .kind/kind-config.yaml
```

Comment:
nit: would be great to read this in from stdin rather than having the config on file. Something like:

```suggestion
kind create cluster --config=-<<-EOF
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
name: spin-test
nodes:
- role: control-plane
  extraPortMappings:
  - containerPort: 80
    hostPort: 8080
    protocol: TCP
EOF

```